OpenRA.Game/Graphics/UISpriteRenderable.cs

```diff
@@ -39,26 +36,26 @@ public UISpriteRenderable(Sprite sprite, WPos effectiveWorldPos, int2 screenPos,
 			// HACK: This is working around the fact that palettes are defined on traits rather than sequences
 			// and can be removed once this has been fixed
 			if (sprite.Channel == TextureChannel.RGBA && !(palette?.HasColorShift ?? false))
-				this.palette = null;
+				Palette = null;
 		}
 
 		// Does not exist in the world, so a world positions don't make sense
-		public WPos Pos => effectiveWorldPos;
+		public WPos Pos { get; }
```

Comment:
More investigating shows that Pos, is an actual world position in some implementations.

Additionally, I found is that `IRenderable.Pos` only exists to be used here

https://github.com/OpenRA/OpenRA/blob/17a2c6a3832221982c7a4b4f3f4b934d850d32ad/OpenRA.Game/Graphics/WorldRenderer.cs#L23-L24

Reply:
The use case is to sort the renderables in the world bottom to top so units and such render above each other correctly. Some renderables such as UI elements don't have the notion of being "in the world", which is where the "effective world pos" comes from. The current naming and comments make sense in this context.

Reply:
Yes, that's what I gathered. This really seems like bad naming. I'm not sure how to go forward with this, I can think of multiple solutions but none seem great

* Add a summary to `IRenderable.Pos` that it's only supposed to be used for calculating Z comparison key
   * This just makes it less confusing, but keeps the shuehorn implementation
* Rename `IRenderable.Pos` to `IRenderable.EffectivePos`
   * Downside is that where it defines real position it becomes misleading
* Refactor so that every `IRenderable` would have to define it's Z comparison key, thus removing `IRenderable.Pos`
   * Downside is that this causes a lot of duplication

Reply:
But we do rename a lot of variables here

Reply:
All the existing properties retain their names. The only "renaming" is when the backing fields disappear - and some have different names to the publically exposed property.